### PR TITLE
Refactor Pixel Fractal effect

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

* Halved array size to reduce memory footprint
* Include `RGB_MATRIX_USE_LIMITS()` 

Fixes https://github.com/qmk/qmk_firmware/issues/17589

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

**Comment by filterpaper** [comment]
Added fix for #17589

File: quantum/rgb_matrix/animations/pixel_fractal_anim.h [hunk 1]
```diff
@@ -22,12 +8,8 @@ RGB_MATRIX_EFFECT(PIXEL_FRACTAL)
 
 static bool PIXEL_FRACTAL(effect_params_t* params) {
 #        define MID_COL MATRIX_COLS / 2
-    static bool led[MATRIX_ROWS][MATRIX_COLS];
-
+    static bool led[MATRIX_ROWS][MID_COL];
```
Comment (drashna) [inline quantum/rgb_matrix/animations/pixel_fractal_anim.h:11]:
```suggestion
    static bool     led[MATRIX_ROWS][MID_COL];
```

File: quantum/rgb_matrix/animations/pixel_fractal_anim.h [hunk 2]
```diff
@@ -37,44 +19,42 @@ static bool PIXEL_FRACTAL(effect_params_t* params) {
...
-            } else {
-                rgb_matrix_set_color(g_led_config.matrix_co[h][l], 0, 0, 0);
+    RGB_MATRIX_USE_LIMITS(led_min, led_max);
+
+    if (g_rgb_timer > wait_timer) {
+
+        RGB rgb = rgb_matrix_hsv_to_rgb(rgb_matrix_config.hsv);
+        for (uint8_t h = 0; h < MATRIX_ROWS; ++h) {
+
+            // Light and copy columns outward
```
Comment (drashna) [inline quantum/rgb_matrix/animations/pixel_fractal_anim.h:29]:
```suggestion
    if (g_rgb_timer > wait_timer) {
        RGB rgb = rgb_matrix_hsv_to_rgb(rgb_matrix_config.hsv);
        for (uint8_t h = 0; h < MATRIX_ROWS; ++h) {
            // Light and copy columns outward
```

**Review by drashna — CHANGES_REQUESTED** [review]
Looks like lint wants some formatting changes:

**Review by drashna — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**